### PR TITLE
Add region tags

### PR DESCRIPTION
**Pull request by PierrickVoulet**

### CODE REVIEW

File: examples/AdvancedOperations/AddSmartDisplayAd.cs [hunk 1]
```diff
@@ -147,6 +149,7 @@ private static string CreateBudget(GoogleAdsClient client, long customerId)
         /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
         /// <param name="budgetResourceName">The campaign budget resource name.</param>
         /// <returns>Resource name of the newly created campaign.</returns>
+        // [START AddSmartDisplayAd_1]
```
Comment (AndrewMBurke) [inline examples/AdvancedOperations/AddSmartDisplayAd.cs:152]: Is there a better/more specific name that we can apply for these examples with multiple region tags? Perhaps including the method name, something like `AddSmartDisplayAd_Run`, `AddSmartDisplayAd_CreateBudget`, etc? I think it'd be a lot easier to maintain the documentation links in the future.

File: examples/AdvancedOperations/AddSmartDisplayAd.cs [hunk 2]
```diff
@@ -310,6 +318,7 @@ private static string CreateBudget(GoogleAdsClient client, long customerId)
         /// <param name="squareMarketingImageAssetResourceName">The square marketing image asset
         /// resource name.</param>
         /// <returns>Resource name of the newly created ad.</returns>
+        // [START AddSmartDisplayAd_3]
```
Comment (AndrewMBurke) [inline examples/AdvancedOperations/AddSmartDisplayAd.cs:321]: Are the order of these numbers coming from the documentation? Again I'd be concerned about maintenance moving forward: which method is "3" or "4" isn't immediately clear, which I'd think would be made more clear if the tag included the method name.

File: examples/AdvancedOperations/UsePortfolioBiddingStrategy.cs [hunk 3]
```diff
@@ -152,6 +154,7 @@ public void Run(GoogleAdsClient client, long customerId)
         /// <param name="name">The budget name.</param>
         /// <param name="amount">The budget amount in micros.</param>
         /// <returns>The budget resource name.</returns>
+        // [START UsePortfolioBiddingStrategy_2]
```
Comment (AndrewMBurke) [inline examples/AdvancedOperations/UsePortfolioBiddingStrategy.cs:157]: Same question regarding tag names and numbering.

File: examples/BasicOperations/AddCampaigns.cs [hunk 4]
```diff
@@ -76,6 +76,7 @@ public void Run(GoogleAdsClient client, long customerId)
 
             for (int i = 0; i < NUM_CAMPAIGNS_TO_CREATE; i++)
             {
+                // [START AddCampaigns_1]
```
Comment (AndrewMBurke) [inline examples/BasicOperations/AddCampaigns.cs:79]: Same question regarding tag names and numbering.

File: examples/CampaignManagement/AddCompleteCampaignsUsingBatchJob.cs [hunk 5]
```diff
@@ -156,6 +158,7 @@ public void Run(GoogleAdsClient client, long customerId)
         /// <param name="batchJobResourceName">The resource name of batch job to which the batch
         /// job operations will be added.
         /// </param>
+        // [START AddCompleteCampaignsUsingBatchJob_1]
```
Comment (AndrewMBurke) [inline examples/CampaignManagement/AddCompleteCampaignsUsingBatchJob.cs:161]: Same question regarding tag names and numbering.

File: examples/ErrorHandling/HandleExpandedTextAdPolicyViolations.cs [hunk 6]
```diff
@@ -180,6 +182,7 @@ private static string[] FetchIgnorablePolicyTopics(GoogleAdsException ex)
         /// <param name="service">The ad group ad service.</param>
         /// <param name="operation">The ad group ad operation to request exemption for.</param>
         /// <param name="ignorablePolicyTopics">The ignorable policy topics.</param>
+        // [START HandleExpandedTextAdPolicyViolations_1]
```
Comment (AndrewMBurke) [inline examples/ErrorHandling/HandleExpandedTextAdPolicyViolations.cs:185]: Same question regarding tag names and numbering.

File: examples/ErrorHandling/HandleKeywordPolicyViolations.cs [hunk 7]
```diff
@@ -216,6 +218,7 @@ private static PolicyViolationKey[] FetchExemptPolicyViolationKeys(GoogleAdsExce
         /// <param name="operation">The ad group criterion operation to request exemption for.
         /// </param>
         /// <param name="exemptPolicyViolationKeys">The exemptable policy violation keys.</param>
+        // [START HandleKeywordPolicyViolations_1]
```
Comment (AndrewMBurke) [inline examples/ErrorHandling/HandleKeywordPolicyViolations.cs:221]: Same question regarding tag names and numbering.

File: examples/ErrorHandling/HandlePartialFailure.cs [hunk 8]
```diff
@@ -138,16 +139,19 @@ private void PrintResults(MutateAdGroupsResponse response)
...
         /// Inspects a response to check for presence of partial failure errors.
         /// </summary>
         /// <param name="response">The response.</param>
         /// <returns>True if there are partial failures, false otherwise.</returns>
+        // [START HandlePartialFailure_1]
         private static bool CheckIfPartialFailureErrorExists(MutateAdGroupsResponse response)
         {
             return response.PartialFailureError == null;
         }
+        // [END HandlePartialFailure_1]
```
Comment (AndrewMBurke) [inline examples/ErrorHandling/HandlePartialFailure.cs:154]: Same question regarding tag names and numbering.

File: examples/Extensions/AddAffiliateLocationExtensions.cs [hunk 9]
```diff
@@ -304,6 +306,7 @@ private static void RemoveFeeds(GoogleAdsClient client, long customerId, Feed[]
         /// <param name="customerId">The customer ID for which the call is made.</param>
         /// <param name="feedResourceName">The feed resource name.</param>
         /// <returns>The newly created feed mapping.</returns>
+        // [START AddAffiliateLocationExtensions_1]
```
Comment (AndrewMBurke) [inline examples/Extensions/AddAffiliateLocationExtensions.cs:309]: Same question regarding tag names and numbering.

File: examples/Extensions/AddGoogleMyBusinessLocationExtensions.cs [hunk 10]
```diff
@@ -344,6 +346,7 @@ private static void RemoveFeeds(GoogleAdsClient client, long customerId, Feed[]
         /// <param name="customerId">The customer ID for which the call is made.</param>
         /// <param name="gmbFeedResourceName">The GMB feed resource name.</param>
         /// <returns>The newly created feed mapping.</returns>
+        // [START AddGoogleMyBusinessLocationExtensions_1]
```
Comment (AndrewMBurke) [inline examples/Extensions/AddGoogleMyBusinessLocationExtensions.cs:349]: Same question regarding tag names and numbering.

File: examples/HotelAds/AddHotelAd.cs [hunk 11]
```diff
@@ -90,6 +90,7 @@ public static void Main(string[] args)
         /// <param name="customerId">The customer ID for which the call is made.</param>
         /// <param name="hotelCenterAccountId">The Hotel Center account ID.</param>
         /// <param name="cpcBidCeilingMicroAmount">The CPC bid ceiling micro amount.</param>
+        // [START AddHotelAd_1]
```
Comment (AndrewMBurke) [inline examples/HotelAds/AddHotelAd.cs:93]: Same question regarding tag names and numbering.

File: examples/Planning/ForecastReach.cs [hunk 12]
```diff
@@ -115,6 +115,7 @@ public void ShowPlannableLocations(ReachPlanServiceClient reachPlanService)
         /// <param name="locationId">Location ID to plan for. To find a valid location ID, either
         /// see https://developers.google.com/adwords/api/docs/appendix/geotargeting or call
         /// <see cref="ReachPlanServiceClient.ListPlannableLocations"/>.</param>
+        // [START ForecastReach_2]
```
Comment (AndrewMBurke) [inline examples/Planning/ForecastReach.cs:118]: Same question regarding tag names and numbering.

File: examples/Remarketing/AddRemarketingAction.cs [hunk 13]
```diff
@@ -95,9 +96,11 @@ public void Run(GoogleAdsClient client, long customerId)
 
                 // Creates a query that retrieves the previously created remarketing action
                 // with its generated tag snippets.
+                // [START AddRemarketingAction]
```
Comment (AndrewMBurke) [inline examples/Remarketing/AddRemarketingAction.cs:99]: Could possibly be `AddRemarketingAction_query`?

File: examples/ShoppingAds/AddShoppingProductAd.cs [hunk 14]
```diff
@@ -174,6 +174,7 @@ private string AddCampaignBudget(GoogleAdsClient client, long customerId)
         /// <returns>Resource name of the newly created campaign.</returns>
         /// <exception cref="GoogleAdsException">Thrown if an API request failed with one or more
         /// service errors.</exception>
+        // [START AddShoppingProductAd_2]
```
Comment (AndrewMBurke) [inline examples/ShoppingAds/AddShoppingProductAd.cs:177]: Same question regarding tag names and numbering.

File: examples/ShoppingAds/AddShoppingSmartAd.cs [hunk 15]
```diff
@@ -177,6 +177,7 @@ private string AddCampaignBudget(GoogleAdsClient client, long customerId)
         /// <returns>Resource name of the newly created campaign.</returns>
         /// <exception cref="GoogleAdsException">Thrown if an API request failed with one or more
         /// service errors.</exception>
+        // [START AddShoppingSmartAd_3]
```
Comment (AndrewMBurke) [inline examples/ShoppingAds/AddShoppingSmartAd.cs:180]: Same question regarding tag names and numbering.

File: examples/ShoppingAds/AddShoppingSmartAd.cs [hunk 16]
```diff
@@ -192,6 +193,7 @@ private string AddCampaignBudget(GoogleAdsClient client, long customerId)
                 MerchantId = merchantCenterAccountId,
             };
 
+            // [START AddShoppingSmartAd]
```
Comment (AndrewMBurke) [inline examples/ShoppingAds/AddShoppingSmartAd.cs:196]: `AddShoppingSmartAd_campaign`?

File: examples/ShoppingAds/ApproveMerchantCenterLink.cs [hunk 17]
```diff
@@ -79,17 +79,21 @@ public void Run(GoogleAdsClient client, long customerId, long merchantCenterAcco
             try
             {
                 // Retrieve all the existing Merchant Center links.
+                // [START ApproveMerchantCenterLink]
```
Comment (AndrewMBurke) [inline examples/ShoppingAds/ApproveMerchantCenterLink.cs:82]: `ApproveMerchantCenterLink_list`?

File: examples/ShoppingAds/ApproveMerchantCenterLink.cs [hunk 18]
```diff
@@ -79,17 +79,21 @@ public void Run(GoogleAdsClient client, long customerId, long merchantCenterAcco
...
                     merchantCenterLinkService.ListMerchantCenterLinks(customerId.ToString());
 
                 Console.WriteLine($"{response.MerchantCenterLinks.Count} Merchant Center link(s)" +
                     $" found with the following details:");
+                // [END ApproveMerchantCenterLink]
 
                 // Iterate the results, and filter for links with pending status.
                 foreach (MerchantCenterLink merchantCenterLink in response.MerchantCenterLinks)
                 {
+                    // [START ApproveMerchantCenterLink_1]
```
Comment (AndrewMBurke) [inline examples/ShoppingAds/ApproveMerchantCenterLink.cs:93]: `ApproveMerchantCenterLink_printStatus`?

File: examples/ShoppingAds/ApproveMerchantCenterLink.cs [hunk 19]
```diff
@@ -119,6 +123,7 @@ public void Run(GoogleAdsClient client, long customerId, long merchantCenterAcco
         /// <param name="merchantCenterLinkService">The merchant center link service.</param>
         /// <param name="merchantCenterLink">The merchant center link.</param>
         /// <param name="status"></param>
+        // [START ApproveMerchantCenterLink_2]
```
Comment (AndrewMBurke) [inline examples/ShoppingAds/ApproveMerchantCenterLink.cs:126]: Same question regarding tag names and numbering.

File: examples/Targeting/AddCampaignTargetingCriteria.cs [hunk 20]
```diff
@@ -194,6 +196,7 @@ public static void Main(string[] args)
         /// </param>
         /// <returns>a campaign criterion object with the specified locationId and resource name.
         /// </returns>
+        // [START AddCampaignTargetingCriteria_1]
```
Comment (AndrewMBurke) [inline examples/Targeting/AddCampaignTargetingCriteria.cs:199]: Same question regarding tag names and numbering.

**Review by AndrewMBurke — COMMENTED** [review]
Mostly the same repeated question regarding naming conventions, just marked each occurrence that I spotted.

File: examples/AdvancedOperations/AddSmartDisplayAd.cs (hunk 1 above)
Comment (PierrickVoulet) [reply examples/AdvancedOperations/AddSmartDisplayAd.cs:152]: I agree and I considered this when writing the script that generated these changes but it was too much effort compared to the benefit. I am happy to discuss this further offline if needed.

File: examples/AdvancedOperations/AddSmartDisplayAd.cs (hunk 2 above)
Comment (PierrickVoulet) [reply examples/AdvancedOperations/AddSmartDisplayAd.cs:321]: This is highly subjective and it was too expensive to write the script to generate more specific names for the scope of this PR. I am happy to discuss this further offline if needed.

Comment (PierrickVoulet) [reply examples/AdvancedOperations/AddSmartDisplayAd.cs:321]: Regarding the number ordering: the numbers are incremented as code snippets are found while crawling the guides that display those.

**Review by AndrewMBurke — APPROVED** [review]

**Review by AnashOommen — APPROVED** [review]
About the autogenerated numbers: I share the concern, but the current approach is a good-enough milestone to start with.

@PierrickVoulet I wonder if we can add some form of autogenerated comment to add a summary of which guide and section this includecode refers to? Perhaps it can be useful for users reading through the examples too, but it has to be done in a way that doesn't make overly verbose comments.

**Comment by PierrickVoulet** [comment]
> About the autogenerated numbers: I share the concern, but the current approach is a good-enough milestone to start with.
> 
> @PierrickVoulet I wonder if we can add some form of autogenerated comment to add a summary of which guide and section this includecode refers to? Perhaps it can be useful for users reading through the examples too, but it has to be done in a way that doesn't make overly verbose comments.

@AnashOommen @AndrewMBurke Something that could be really useful and I believe fairly easy to generate and maintain would be a list of the guides that use code snippets from this code example. Typically, we could periodically regenerate a header comment for each code example with something like the following:

```
You can learn more about what this code example is used for by reading the following guides:
- [Change Status service](https://developers.google.com/google-ads/api/docs/change-status)
- ...
```

Any thoughts?

**Pull request merged**